src/main/kotlin/io/snyk/plugin/snykcode/SnykCodeBulkFileListener.kt

```diff
@@ -11,51 +13,66 @@ import com.intellij.openapi.vfs.VirtualFileManager
 import com.intellij.openapi.vfs.newvfs.events.VFileEvent
 import com.intellij.openapi.vfs.readText
 import io.snyk.plugin.SnykBulkFileListener
-import io.snyk.plugin.SnykFile
 import io.snyk.plugin.getSnykCachedResults
 import io.snyk.plugin.toLanguageServerURL
 import io.snyk.plugin.toSnykFileSet
 import org.eclipse.lsp4j.DidSaveTextDocumentParams
 import org.eclipse.lsp4j.TextDocumentIdentifier
 import snyk.common.lsp.LanguageServerWrapper
+import java.time.Duration
 
 class SnykCodeBulkFileListener : SnykBulkFileListener() {
+    // Cache for debouncing file updates that come in within one second of the last
+    // Key = path, Value is irrelevant
+    private val debounceFileCache =
+        CacheBuilder.newBuilder()
+            .expireAfterWrite(Duration.ofMillis(1000)).build<String, Boolean>()
+
     override fun before(project: Project, virtualFilesAffected: Set<VirtualFile>) = Unit
 
     override fun after(project: Project, virtualFilesAffected: Set<VirtualFile>) {
```

Comment:
Do we need similar changes in https://github.com/snyk/snyk-intellij-plugin/blob/1b78014fbf3f8810e35ef7ce7d48e5a63ffbbf02/src/main/kotlin/snyk/oss/OssBulkFileListener.kt?

Reply:
I think it would be good to have some re-usable code for this but I think we can leave that for when we finish the OSS in LS pitch next cycle right?

Reply:
Yeah, I usually only do this kind of thing, the second time we need it :)